Build failure: notion 4.0.4

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Try and build a system depending on Notion (my system-path depends on it)

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317880053

### Relevant log output

```console
error: Cannot build '/nix/store/aa9f2x3qfrdj7jpfq0chpwdr1djmsb10-notion-4.0.4.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/3ng232jibdqs3f6d8a6dnkaapvccsyfb-notion-4.0.4-man
         /nix/store/bf2mgdyjks1hzylbjl8xk7x7ymn608mb-notion-4.0.4
       Last 25 log lines:
       >    29 |     void (*destroy_fn)();
       >       |            ^~~~~~~~~~
       > obj.c: In function 'comp_fun':
       > obj.c:130:14: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   130 |     void *af=(void*)((DynFunTab*)a)->func;
       >       |              ^
       > obj.c:131:14: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   131 |     void *bf=(void*)((DynFunTab*)b)->func;
       >       |              ^
       > obj.c: In function 'lookup_dynfun':
       > obj.c:192:20: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   192 |                 if((void*)df[ndx].func==(void*)func){
       >       |                    ^
       > obj.c:192:41: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   192 |                 if((void*)df[ndx].func==(void*)func){
       >       |                                         ^
       > obj.c:196:20: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   196 |                 if((void*)df[ndx].func<(void*)func)
       >       |                    ^
       > obj.c:196:40: warning: ISO C forbids conversion of function pointer to object pointer type [-Wpedantic]
       >   196 |                 if((void*)df[ndx].func<(void*)func)
       >       |                                        ^
       > make[1]: *** [<builtin>: obj.o] Error 1
       > make[1]: Leaving directory '/build/source/libtu'
       > make: *** [build/rules.mk:175: subdirs] Error 2
       For full logs, run:
         nix log /nix/store/aa9f2x3qfrdj7jpfq0chpwdr1djmsb10-notion-4.0.4.drv
copying path '/nix/store/ycfz7a4jwm7084c8c44h1rvnz66zlh8q-gcc-wrapper-15.2.0' from 'https://cache.nixos.org'
copying path '/nix/store/5g03z46bx02591l5l5dg40pwhrd0fcpw-libjack2-1.9.22' from 'https://cache.nixos.org'
copying path '/nix/store/4ln27npm1dhw76ncvbj9z38vnk8y1zgv-libpulseaudio-17.0' from 'https://cache.nixos.org'
copying path '/nix/store/pia3knz0d0fwmx13f8hkzc0rnhw7z8xw-man-db-2.13.1' from 'https://cache.nixos.org'
copying path '/nix/store/2y5x1l6mwllm1l0ly6z3vmhlnpnghy5r-mupdf-1.26.10' from 'https://cache.nixos.org'
copying path '/nix/store/43ar6s3la8ba0jv2mxf8ym482f46s0s8-marksman-2025-12-13' from 'https://cache.nixos.org'
copying path '/nix/store/blivz687snlf73vr1zbb0bcr1k3c6vfm-nix-fetchers-2.30.3+1' from 'https://cache.nixos.org'
error: Cannot build '/nix/store/9c7bpvd0ldaq76nqdcwkhcwhh25pq7hp-system-path.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/i6m7p070sa9qqnawj652ylfg4bdk8kfd-system-path
error: Cannot build '/nix/store/x6cdr2rq30kz3siafl5cz5a48pwa64rr-nixos-system-mainSystem-26.05.20260116.e4bae1b.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/73dm3cpna34vkbkrncqp56ygv8bza5y8-nixos-system-mainSystem-26.05.20260116.e4bae1b
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.63, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`

### Notify maintainers

@raboof @NotAShelf 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/no/notion/package.nix
+++ b/pkgs/by-name/no/notion/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   fontconfig,
   gettext,
   groff,
@@ -29,6 +30,14 @@ stdenv.mkDerivation (finalAttrs: {
     tag = finalAttrs.version;
     hash = "sha256-L7WL8zn1Qkf5sqrhqZJqFe4B1l9ULXI3pt3Jpc87huk=";
   };
+
+  patches = [
+    # GCC 15 fix
+    (fetchpatch2 {
+      url = "https://github.com/raboof/notion/commit/89c92f49abfeae1168ad343d4f529a52d0edd78c.patch?full_index=1";
+      hash = "sha256-+4GGeY2j7B54Ffw5gFNpG4704Egc7rA6w5z0sZG8210=";
+    })
+  ];
 
   nativeBuildInputs = [
     gettext

